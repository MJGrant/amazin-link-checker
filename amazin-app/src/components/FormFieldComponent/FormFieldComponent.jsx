import React, { Component } from 'react'
import PropTypes from 'prop-types'

import { Label, Input, FormGroup, Row, Col, FormFeedback } from 'reactstrap'

export default class FormFieldComponent extends Component {
    static propTypes = {
        fieldID: PropTypes.string,
        errorClass: PropTypes.string,
        nameVal: PropTypes.string,
        val: PropTypes.string,
        placeholderText: PropTypes.string
    }

    static defaultProps = {
        fieldID: '',
        labelContents: '',
        errorClass: '',
        nameVal: '',
        val: '',
        placeholderText: ''
    }

    render() {
        const { fieldID, labelContents, errorClass, nameVal, val, handleChange, handleBlur, feedbackText, placeholderText } = this.props;
        return (
            <Row form>
                <Col lg={12}>
                    <FormGroup>
                        <Label for={fieldID}>{labelContents}</Label>
                        <Input
                            className={errorClass}
                            type="text"
                            name={nameVal}
                            value={val}
                            onChange={handleChange}
                            onBlur={handleBlur}
                            id={fieldID}
                            placeholder={placeholderText} />
                        <FormFeedback>{feedbackText}</FormFeedback>
                    </FormGroup>
                </Col>
            </Row>
        )
    }
}

