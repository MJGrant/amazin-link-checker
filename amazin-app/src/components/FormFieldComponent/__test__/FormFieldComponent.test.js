import React from 'react'
import renderer from 'react-test-renderer'

import FormFieldComponent from '../FormFieldComponent'

const handleChange = jest.fn();
const handleBlur = jest.fn();

describe('FormFieldComponent', () => {
    test('renders itself with defaults', () => {
        const component = renderer.create(<FormFieldComponent />)
        expect(component).toMatchSnapshot()
    })

    test('renders itself text prop', () => {
        const component = renderer.create(<FormFieldComponent
            labelContents="Test label here"
            errorClass=""
            nameVal="http://localhost.test/"
            val="test-val"
            fieldID="test-id"
            handleChange={handleChange}
            handleBlur={handleBlur}
            placeholderText="http://localhost.test/yourarticle"
            feedbackText="Test feedback text here"
        />)
        expect(component).toMatchSnapshot()
    })
})