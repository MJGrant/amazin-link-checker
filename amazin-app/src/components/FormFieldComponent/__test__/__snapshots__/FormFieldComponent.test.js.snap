// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormFieldComponent renders itself text prop 1`] = `
<div
  className="form-row"
>
  <div
    className="col-lg-12"
  >
    <div
      className="form-group"
    >
      <label
        className=""
        htmlFor="test-id"
      >
        Test label here
      </label>
      <input
        className="form-control"
        id="test-id"
        name="http://localhost.test/"
        onBlur={[MockFunction]}
        onChange={[MockFunction]}
        placeholder="http://localhost.test/yourarticle"
        type="text"
        value="test-val"
      />
      <div
        className="invalid-feedback"
      >
        Test feedback text here
      </div>
    </div>
  </div>
</div>
`;

exports[`FormFieldComponent renders itself with defaults 1`] = `
<div
  className="form-row"
>
  <div
    className="col-lg-12"
  >
    <div
      className="form-group"
    >
      <label
        className=""
        htmlFor=""
      >
        
      </label>
      <input
        className="form-control"
        id=""
        name=""
        placeholder=""
        type="text"
        value=""
      />
      <div
        className="invalid-feedback"
      />
    </div>
  </div>
</div>
`;
